tracing-attributes/tests/fields.rs

```diff
@@ -146,6 +149,40 @@ fn string_field() {
     });
 }
 
+#[test]
+fn keyword_ident_in_field_name() {
+    let span = expect::span().with_fields(
+        expect::field("keywords.impl.type.fn")
+            .with_value(&"test")
+            .only(),
+    );
+    run_test(span, || fn_keyword_ident_in_field("test"));
+}
+
+#[test]
+fn keyword_ident_in_field_name_event_macro() {
+    let (collector, handle) = collector::mock()
+        .event(expect::event().with_fields(expect::field("crate").with_value(&"tracing")))
+        .only()
+        .run_with_handle();
+
+    with_default(collector, || error!(crate = "tracing", "message"));
+    handle.assert_finished();
+}
+
+#[test]
+fn keyword_ident_in_field_name_span_macro() {
+    #[derive(Debug)]
+    struct Foo;
+
+    let span =
+        expect::span().with_fields(expect::field("self").with_value(&field::debug(Foo)).only());
+    run_test(span, || {
+        let guard = error_span!("span", self = ?Foo).entered();
+        drop(guard);
+    });
+}
+
```

Comment:
I think that these tests should probably be in `tracing/tests/event.rs` and `tracing/tests/span.rs` respecivetly, since they're not testing anything from the `tracing-attributes` crate.

Reply:
Aha, that's why I didn't find any existing tests for them ^^

Reply:
Done, but they're failing for me locally now... Same as any other tests I try to run from one of those files :shrug:

Reply:
What error are you getting?

Reply:
I got "more notifications expected" and can reproduce the problem on the commandline (previously was RA) by passing `--all-features`, e.g. `cargo test --test event --all-features`.

Reply:
Using `--all-features` with `tracing` does all sorts of things that you don't want for those tests. There are features that statically disable traces (e.g. https://github.com/tokio-rs/tracing/blob/master/tracing/Cargo.toml#L51) and that plays havoc with tests. Generally I think you don't want any features enabled unless they're explicitly needed for that test